src/test/java/com/uber/cadence/workflow/workflows/SendSignalObject_WorkflowImpl.java

```diff
@@ -0,0 +1,36 @@
+/*
+ *  Copyright 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not
+ *  use this file except in compliance with the License. A copy of the License is
+ *  located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on
+ *  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package com.uber.cadence.workflow.workflows;
```

Comment:
Minor comment: One implementation per file might be too verbose. It might make sense to keep a set of related implementations in one file.

Reply:
Do we have some sort of rough guidance on how we organize our classes/files? i'm used to one class per file. Relevant ones live in the same directory then they're collapsable which keeps the the noise in the project view low. At the same time content of the file and available classes are obvious from just expanding and looking at that directory

Reply:
The default is one class per file.  But if some class is used only as an implementation helper for another class then it is usually embedded in that class. 
Unit tests tend to embed all classes that the given unit test needs.

Reply:
We have multiple tests living in one file. With every test adding it's own helper methods/classes as members of the same test class, the class grows really large. It also discourages visibility and reuse as new tests get written. 
LoggerTest.TestWorkflow, MetricsTest.TestWorkflow1, MetricsTest.TextExceptionPropagation, MetricsTest.ITestChild interfaces look the same.

Reply:
I see no problem having single MetricTest.Workflow interface instead of three. But at the same time there is not that much code duplication even now. I don't think that moving all of these dozens of workflows that are used by a single test into separate files going to improve visibility.

Reply:
moved the classes into the test file